src/pymongo_migrate/cli.py

```diff
@@ -59,6 +60,18 @@ def get_logger(verbose: int):
 def mongo_migrate_decor(f):
     @wraps(f)
     def wrap_with_client(uri, migrations, collection, verbose, *args, **kwargs):
+        # Percent-encode the username and password, if provided.
+        uri = urllib.parse.urlparse(uri)
+        if uri.username:
+            uri = uri._replace(
+                netloc="%s:%s@%s"
+                % (
+                    urllib.parse.quote_plus(uri.username),
+                    urllib.parse.quote_plus(uri.password),
+                    uri.hostname,
+                ),
+            )
+        uri = urllib.parse.urlunparse(uri)
```

Comment:
it would be nice to put it in a function, especially as we need some tests for this as well

Reply:
Working on it, this is just a draft :-)